src/Servers/Kestrel/Core/src/Internal/Infrastructure/PipeWriterHelpers/ConcurrentPipeWriter.cs

```diff
@@ -57,6 +57,10 @@ public ConcurrentPipeWriter(PipeWriter innerPipeWriter, MemoryPool<byte> pool, o
         _sync = sync;
     }
 
+    public override bool CanGetUnflushedBytes => true;
+
+    public override long UnflushedBytes => _currentFlushTcs is null && _head is null ? _innerPipeWriter.UnflushedBytes : _bytesBuffered;
```

Comment:
Shouldn't this be?

```suggestion
    public override long UnflushedBytes => _currentFlushTcs is null && _head is null ? _innerPipeWriter.UnflushedBytes : _bytesBuffered + _innerPipeWriter.UnflushedBytes;
```

Also, I don't think `ConcurrentPipeWriter` is guaranteed to have an `_innerPipeWriter` where `CanGetUnflushedBytes` is true. For `Http1Connection,` it comes from `ConnectionContext.Transport`.

Reply:
If it should be this, and the current code passes all tests, then a test should be added to verify correctness here.

Reply:
Removed it